slither/detectors/functions/external_function.py

```diff
@@ -135,6 +151,14 @@ def _detect(self):  # pylint: disable=too-many-locals,too-many-branches
             # Next we'll want to loop through all functions defined directly in this contract.
             for function in contract.functions_declared:
 
+                # If all of the function arguments are non-reference type or calldata, we skip it.        
+                reference_args = []
+                for arg in function.parameters:
+                    if self.is_reference_type(arg) and arg.location == 'memory':
+                        reference_args.append(arg)
+                if len(reference_args) == 0 and len(function.parameters) > 0:
```

Comment:
What's the reasoning behind this check `len(function.parameters) > 0` ?

Reply:
Ops, my mistake! We can skip functions without arguments there too. I removed the check in the next commit.